### PR TITLE
RavenDB-5851: RavenDB Server on MacOsx

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Sparrow/Platform/Posix/PerPlatformValues.cs [hunk 1]
```diff
@@ -16,6 +17,7 @@ public class SyscallNumbers
         public class OpenFlags
         {
             public static Posix.OpenFlags O_DIRECT = (Posix.OpenFlags) (
+                RunningOnMacOsx ? 0 :
```
Comment (ayende) [inline src/Sparrow/Platform/Posix/PerPlatformValues.cs:20]: Comment here, explaining that it is not supported.

File: src/Sparrow/Platform/Posix/PerPlatformValues.cs [hunk 2]
```diff
@@ -28,10 +30,39 @@ public class OpenFlags
                 : 65536); // value directly from printf("%d", O_DIRECTORY) on the pi
 
             public static Posix.OpenFlags O_LARGEFILE = (Posix.OpenFlags)(
+                RunningOnMacOsx ? 0 :
```
Comment (ayende) [inline src/Sparrow/Platform/Posix/PerPlatformValues.cs:33]: comment here, explaining not supported and why

File: src/Sparrow/Platform/Posix/PosixSodium.cs [hunk 3]
```diff
@@ -32,90 +38,95 @@ static PosixSodium()
...
             }
         }
 
         // ReSharper disable once InconsistentNaming
         public static UIntPtr crypto_generichash_bytes()
         {
-            return _is32bits ?
-                (_isArm ? Arm.crypto_generichash_bytes() : X86.crypto_generichash_bytes()) :
-                X64.crypto_generichash_bytes();
+            return _is32bits ? (_isArm ? Arm.crypto_generichash_bytes() : X86.crypto_generichash_bytes()) : _isMac64 ? MacOsxX64.crypto_generichash_bytes() : X64.crypto_generichash_bytes();
```
Comment (ayende) [inline src/Sparrow/Platform/Posix/PosixSodium.cs:67]: @redknightlois Can you think of any better way to handle this?

File: src/Sparrow/Platform/Posix/PosixSodium.cs [hunk 4]
```diff
@@ -1308,5 +1331,282 @@ public static void Initialize()
...
+                try
+                {
+                    var rc = sodium_init();
+                    if (rc != 0)
+                        throw new InvalidOperationException("Unable to initialize sodium, error code: " + rc);
+                }
+                catch (DllNotFoundException dllNotFoundEx)
+                {
+                    // make sure the lib file is not there (this exception might pop when incorrect libsodium lib is does exists)
+                    if (File.Exists(LIB_SODIUM))
```
Comment (ayende) [inline src/Sparrow/Platform/Posix/PosixSodium.cs:1350]: Shouldn't we provide some more instruction here?

File: src/Sparrow/Platform/Posix/PosixSodium.cs (hunk 3 above)
Comment (redknightlois) [reply src/Sparrow/Platform/Posix/PosixSodium.cs:67]: Cost is going to be dominated by hash and marshalling. The structure of that code is going to be very packed. There is hardly any improvement that can be made unless we are going to do some bond like stuff that would only make sense if we decide to tackle the most general issue.

**Pull request merged**